### DISCUSSION TITLE
SSL Protocol Error - Turkey

### DISCUSSION THREAD

**Discussion by yvzfth** [vercel/community#5102] [Help] [answered]
### Summary

Vercel app domains give the "err_ssl_protocol_error" but the SSL certificate not expired yet. It happened all other my projects too.
<img width="539" alt="Ekran Resmi 2023-12-07 11 24 59" src="https://github.com/vercel/community/assets/70853249/b2eb39cd-dcba-4968-8910-26af75a7306a">
You can see expiring date 24 Dec 2023. 
What should I do? 

### Example

https://bioproftest.vercel.app/

### Steps to Reproduce

![WhatsApp Image 2023-12-07 at 11 36 05](https://github.com/vercel/community/assets/70853249/52ba8614-2e46-4726-9414-dd4391186456)

**Comment by matricore** [comment]
I think Vercel Server has been blocked by Turkish ISP named TURK TELEKOM :(

**Reply by osmanozbatir** [reply]
Not only Turk Telekom but also other ISPs are blocked it. This might be a BTK block.

**Reply by acsiseci** [reply]
I thought I made a mistake. I was very happy when I saw this post :) I hope they fix it soon

**Reply by tolgaouz** [reply]
I was going crazy not being able to access any of my preview deployments 🤡 . Thank you for this

**Comment by swarnava** [comment]
Hi,

This looks like indeed ISP level block indeed and we are investigating currently.

**Reply by osmanozbatir** [reply]
Any updates?

**Reply by swarnava** [reply]
No updates yet. We are in touch with ISP and yet to hear back!

**Reply by ekimcem** [reply]
We also have the same issue @yvzfth but if BTK blocked, how can I reach the website with my mobile internet ( using Turkcell).
In teknokent we asked IT to talk with providers and we are also still waiting answer from them.

**Reply by yvzfth** [reply]
@ekimcem you can use VPN, there is no update yet. Also you can use Superonline ISP if you find in your nearby as I use, it works well.

**Reply by snndmnsz** [reply]
@yvzfth   I have Superonline (Turkcell home internet) as my ISP, and I'm facing issues with the connection as I can't access the website using it. Interestingly, when I tried using Turkcell cellular data, I could access the website without any problems.

**Reply by mustafaselman** [reply]
turk telekom home and phone and also vodafone didnt work

**Reply by osmanozbatir** [reply]
This is really weird. I use Vodafone on my mobile phone and I cannot access the sites from my mobile phone. When I share my connection with my computer, it opens blocked sites without any problem.

**Comment by mustafaselman** [comment]
I have a same problem please help vercel community

**Comment by yeg2799** [comment]
Same problem.

**Comment by swarnava** [comment]
We have already reached out to ISP provider via our internal channel. Unfortunately, we need to wait till we hear back from the team.

**Comment by ouzaltun** [comment]
I have been experiencing the same issue for 2-3 days. I am waiting for your solution suggestions.

**Reply by osmanozbatir** [reply]
I asked to cimer why and which provider blocked the access. Suggest you to do the same thing. If they get too many requests it may speed up the process.

**Reply by ekimcem** [reply]
I also created a ticket there, they will not response less than a month I think, the ISPs should investigate the issue..

**Comment by serter1** [comment]
The problem continues

**Comment by acsiseci** [comment]
My solution is to redirect to a subdomain until the error is fixed :(

**Comment by riobits** [comment]
I have the same problem.

**Reply by riobits** [reply]
They solved it! 🎉

**Comment by haktanonur** [comment]
I also have the same problem.

**Reply by concman** [reply]
You can access the preview deployments through VPN.

**Comment by emrecoban** [comment]
@swarnava says that:
> Hi,
> 
> This looks like indeed ISP level block indeed and we are investigating currently.

But there seems **to be another problem**, because there is still **no legal decision** about Vercel. Here is what the site information query page shows:

![vercel-com](https://github.com/vercel/community/assets/4332481/5fb8bb1e-4fe8-4508-8b75-8d4ff4fe888c)
![vercel-app](https://github.com/vercel/community/assets/4332481/33ea08ef-2984-499c-81bb-c88a9d6b66b6)

I am unable to access my applications through vercel.app on my phone internet (Vodafone), but they can be accessed via a custom domain.

**Reply by can-erturk** [reply]
Apps with custom domains are also not accessible.

**Reply by emrecoban** [reply]
@can-erturk may I ask you to try [https://emre.run](https://emre.run)?

**Reply by can-erturk** [reply]
@emrecoban sorunsuz çalışıyor. merak ettim de, vercel üzerinde mi acaba?

**Reply by emrecoban** [reply]
@can-erturk tabii ki, bilgilendirme için teşekkür ederim.

![vercel-dashboard](https://github.com/vercel/community/assets/4332481/a12c4266-8be3-44aa-83da-9311f87bbf9a)

Hi @smaeda-ks, I can reach my app with a custom domain name, but not vercel.app. Perhaps this information might be useful to you. I use Google Domains.

**Comment by Dazzlin-32** [comment]
I also faced the same problem when I tried to deploy a new project with vercel, but my other projects are working fine. I used vpn to access my new deployment and it was working well. The problem is still not fixed tho.

**Comment by serter1** [comment]
When I look at it now, the problem in my projects seems to have disappeared.

**Comment by 0x74h51N** [comment]
## Individual and Temporary Solution
You can access your site deployed on Vercel without encountering SSL errors by using [DNS Over HTTPS settings](https://www.technopat.net/2023/02/26/oblivious-dns-over-https-nedir/).

Another solution is to use a VPN, but it's clear that you won't get sufficient speed from free VPN services.

## Massive and Permanent Solution
Let's start a riot...

**Comment by yusuferkoc** [comment]
The problem continues

**Comment by bulutyerli** [comment]
I had to move my websites from vercel. Thats really bad.

**Comment by smaeda-ks** [comment, marked as answer]
We have yet to hear back from the ISP. Though there can be multiple ways that NSPs (Network Service Providers) may block your traffic.

### 1. DNS filtering
- To confirm this, run `dig example.com` (replace `example.com` with your domain name) on the same machine/network and see the answer. If your domain isn't resolving to one of our IP ranges (`76.76.21.0/24`), then it's highly likely someone in the middle, such as your ISP, is intercepting your DNS query and returning a wrong answer, causing clients to access a different server that can fail to establish a secure connection.
- You may be able to get around this by using public DNS services, such as Google DNS.

### 2. IP blocking
- To confirm, run `ping 76.76.21.21` or `ping cname.vercel-dns.com` and see if the ping timeouts. If you see timeouts, your ISP might be blocking our IP range `76.76.21.0/24`.
- You may be able to get around this by using VPN services.

### 3. DPI (Deep Packet Inspection) blocking
- NSPs may inspect your traffic using DPI by checking your TLS packet, commonly for TLS ClientHello packets, to determine which domain name the client is trying to access. By doing so, they may block access to certain domains by their own rules.
- You may be able to get around this by using VPN services or by enabling DNS over HTTPS on your browser.

### 4. Advanced blocking by your home router
- Some modern home network routers implement advanced security features by default. These features are usually backed by third-party threat intelligence providers who can inspect traffic using more sophisticated technology than the traditional DPI method mentioned above.
- You may be able to get around this by disabling such security features in your home network router device.

Based on the reports so far, it's either 3 or 4. Regardless, we will let you know when we have an update to share. We're sorry for the trouble. Unfortunately, working with NSPs isn't always very straightforward due to each region's different internet regulations (including censorship). We will do what we can do at best.

Meanwhile, please also let us know when you see signs of recovery.

**Reply by smaeda-ks** [reply]
@emrecoban @osmanozbatir could you collect https://gist.github.com/smaeda-ks/5997018d5fe5dce44d047299829288e1 and get back to me with the full result? (e.g., share by a gist link)
If you are concerned about your privacy, you can DM me: https://x.com/jrsyo

**Reply by emrecoban** [reply]
I've sent the gist link via X due to the privacy. Btw, it doesn't work on Vodafone.

**Reply by osmanozbatir** [reply]
@swarnava fixed for me as well thank you

**Reply by emrecoban** [reply]
The issue is resolved. Happy code everyone!

**Comment by can-erturk** [comment]
@smaeda-ks I tried on multiple devices and I can access vercel.app domains. It seems like the problem has been resolved.

**Reply by osmanozbatir** [reply]
I still can not access vercel.app domains.
ISP: Turk Telekom

**Reply by can-erturk** [reply]
Vodafone and Turksat ISPs work without any problems for me.

**Reply by riobits** [reply]
Superonline works fine for me.

**Reply by Dazzlin-32** [reply]
Both Vodafone and turktelekom are not working for me

**Reply by bulutyerli** [reply]
currently superonline works fine

**Reply by yusufusta** [reply]
Turk Telekom works fine for me

**Reply by ekimcem** [reply]
Superonline & Turk Telekom is tested in 5 different sources, they work well.

**Reply by osmanozbatir** [reply]
It seems solved for my end too. I can access via Turk Telekom.

**Comment by mustafaselman** [comment]
Thanks its work for now.

**Comment by fossejc** [comment]
It seems like I have the same issue with the provider Swisscom in Switzerland.

I can access our website from any network but not from the Swisscom one. It is the main internet provider here in Switzerland and so it is a big issue for us.

Is there anything I can do ?

**Comment by ulasakdeniz** [comment]
Hi, I think it is blocked in Turkey again. DNS resolution doesn't work. Nothing to do technically if it is blocked just letting everyone know :)

**Reply by ulasakdeniz** [reply]
@swarnava I noticed last time you said you investigated it, can you maybe check?

**Reply by ulasakdeniz** [reply]
I can confirm that it is blocked. https://internet.btk.gov.tr/sitesorgu/

<img width="1019" alt="Screenshot 2024-03-27 at 16 49 01" src="https://github.com/vercel/community/assets/5461082/ff3c338c-a61e-4c34-a597-d4faf796d746">

**Reply by furkancakici** [reply]
My country did'nt surprise again :/

**Reply by CllDMR-ekonobi** [reply]
Yeah, again... Another BS

**Reply by NyllRE** [reply]
uuuuuh brother why is turkey doing this

**Reply by ekimcem** [reply]
Well the name server vercel.app is blocked, if you connect your own domain to your vercel app it will work fine for now.

**Reply by smaeda-ks** [reply]
The Turkish regulator has unblocked it. Unfortunately, this is one of those regions with a unique internet law, and it is not very easy to prevent this from happening.

**Reply by ulasakdeniz** [reply]
That's a very professional way to put it @smaeda-ks :)  thanks for the update.
